### PR TITLE
Allow to access APISpec components in plugin init_spec

### LINKED ISSUE #538: Ability to access APISpec.components in plugin init_spec method
Is there a specific reason why components is initialized after plugins init_spec methods are called?

If not, then I am willing to open a PR to solve this issue.

### PR DESCRIPTION
**Pull request by Colin-b**
closes #538 

Note that I tried to manually follow flake8 as my corporate network is using a self signed certificate to perform "man in the middle" preventing Git to access flake8 repo on gitlab (so pre-commit was failing).

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -276,6 +276,19 @@ def test_to_yaml(self, spec):
...
+    def test_components_can_be_accessed_by_plugin_in_init_spec(self):
+        class TestPlugin(BasePlugin):
+            def init_spec(self, spec):
+                spec.components.schema(
+                    "TestSchema", {"properties": {"key": {"type": "string"}}, "type": "object"}
+                )
+
+        spec = APISpec("Test API", version="0.0.1", openapi_version="2.0", plugins=[TestPlugin()])
+        metadata = spec.to_dict()
+        assert metadata["definitions"] == {
```
Comment (lafrech) [inline tests/test_core.py]: Use get_schema form tests/utils.py.

**Review by lafrech — CHANGES_REQUESTED** [review]
AFAIU, flake8 is fine. The CI issues are with black.

I don't understand your setup but I suppose you can pip install flake8 black and run them. Or just apply manually the changes in there: https://dev.azure.com/sloria/sloria/_build/results?buildId=2793&view=logs&j=7e27d0d1-c3b5-5589-c3a3-eb2d9b518da8&t=f9c3c3ee-9bfd-5d70-3f98-d6c193ea86e4.

Or wait until I get the time to do it myself.

Otherwise, this is fine. Just a nitpick in the test.

File: tests/test_core.py (hunk 1 above)
Comment (Colin-b) [reply tests/test_core.py]: Done :)

**Comment by Colin-b** [comment]
> AFAIU, flake8 is fine. The CI issues are with black.
> 
> I don't understand your setup but I suppose you can pip install flake8 black and run them. Or just apply manually the changes in there: https://dev.azure.com/sloria/sloria/_build/results?buildId=2793&view=logs&j=7e27d0d1-c3b5-5589-c3a3-eb2d9b518da8&t=f9c3c3ee-9bfd-5d70-3f98-d6c193ea86e4.
> 
> Or wait until I get the time to do it myself.
> 
> Otherwise, this is fine. Just a nitpick in the test.

I manually applied flake8 before but I was unsure something else was performed by pre-commit lib.
I applied black as well and it refactored the test :)

**Review by lafrech — APPROVED** [review]

**Pull request merged**